From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 15 May 2026 00:00:00 -0400
Subject: [PATCH] Add missing absl base/container/functional/hash/status targets

TF 2.21.0 / XLA reference absl targets the systemlib BUILD files did not
declare: base:no_destructor, container:hash_container_defaults,
functional:overload, hash:hash_testing, status:status_matchers. All are
header-only in abseil; declare them with the appropriate deps.
---
diff --git a/third_party/xla/third_party/absl/system.absl.base.BUILD b/third_party/xla/third_party/absl/system.absl.base.BUILD
index 3c4e3dab..159f87e6 100644
--- a/third_party/xla/third_party/absl/system.absl.base.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.base.BUILD
@@ -139,3 +139,11 @@ cc_library(
         "//absl:__subpackages__",
     ],
 )
+
+cc_library(
+    name = "no_destructor",
+    deps = [
+        ":config",
+        ":nullability",
+    ],
+)
diff --git a/third_party/xla/third_party/absl/system.absl.container.BUILD b/third_party/xla/third_party/absl/system.absl.container.BUILD
index 95c16260..cbcd0d02 100644
--- a/third_party/xla/third_party/absl/system.absl.container.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.container.BUILD
@@ -110,6 +110,14 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "hash_container_defaults",
+    deps = [
+        ":hash_function_defaults",
+        "//absl/base:config",
+    ],
+)
+
 cc_library(
     name = "hash_policy_traits",
     deps = ["//absl/meta:type_traits"],
diff --git a/third_party/xla/third_party/absl/system.absl.functional.BUILD b/third_party/xla/third_party/absl/system.absl.functional.BUILD
index 9439bd0b..6fbf8f40 100644
--- a/third_party/xla/third_party/absl/system.absl.functional.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.functional.BUILD
@@ -13,3 +13,11 @@ cc_library(
 cc_library(
     name = "function_ref",
 )
+
+cc_library(
+    name = "overload",
+    deps = [
+        "//absl/base:config",
+        "//absl/meta:type_traits",
+    ],
+)
diff --git a/third_party/xla/third_party/absl/system.absl.hash.BUILD b/third_party/xla/third_party/absl/system.absl.hash.BUILD
index 4b10eb64..2c7c6a86 100644
--- a/third_party/xla/third_party/absl/system.absl.hash.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.hash.BUILD
@@ -29,3 +29,14 @@ cc_library(
     name = "low_level_hash",
     # absl_low_level_hash was merged into absl_hash in 20260107
 )
+
+cc_library(
+    name = "hash_testing",
+    testonly = True,
+    deps = [
+        "//absl/meta:type_traits",
+        "//absl/strings",
+        "//absl/types:variant",
+        "@com_google_googletest//:gtest",
+    ],
+)
diff --git a/third_party/xla/third_party/absl/system.absl.status.BUILD b/third_party/xla/third_party/absl/system.absl.status.BUILD
index e50e9790..7a09df7c 100644
--- a/third_party/xla/third_party/absl/system.absl.status.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.status.BUILD
@@ -29,3 +29,15 @@ cc_library(
         "//absl/utility",
     ],
 )
+
+cc_library(
+    name = "status_matchers",
+    testonly = True,
+    deps = [
+        ":status",
+        ":statusor",
+        "//absl/base:config",
+        "//absl/strings:string_view",
+        "@com_google_googletest//:gtest",
+    ],
+)
